firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -31,16 +33,46 @@ private CrossProcessLock(FileChannel channel, FileLock lock) {
     this.lock = lock;
   }
 
+  /**
+   * Create and return a lock that is exclusive across processes. If another process
+   * has the lock then this call will block until it is released.
+   * @param lockName the lockname is global across all processes of an app
+   * @return a CrossProcessLock if success. If the lock failed to acquire (maybe due to disk full
+   * or other unexpected and unsupported permissions) then null will be returned.
+   */
   static CrossProcessLock acquire(Context appContext, String lockName) {
+    FileChannel channel = null;
+    FileLock lock = null;
     try {
       File file = new File(appContext.getFilesDir(), lockName);
-      FileChannel channel = new RandomAccessFile(file, "rw").getChannel();
+      channel = new RandomAccessFile(file, "rw").getChannel();
       // Use the file channel to create a lock on the file.
       // This method blocks until it can retrieve the lock.
-      FileLock lock = channel.lock();
-      return new CrossProcessLock(channel, lock);
+      lock = channel.lock();
+      CrossProcessLock crossProcessLock = new CrossProcessLock(channel, lock);
+      channel = null;
+      lock = null;
```

Comment:
Do we need this?
Can't we just do this:

      return crossProcessLock;
    } catch (IOException e) {
      Log.e(TAG, "encountered error while creating and acquiring the lock, ignoring", e);
      if (channel == null) {
        return null;
      }
      try {
        channel.close();
      } catch (IOException e) {
        // nothing to do here
      }
      if (lock != null) {
        try {
          lock.release();
        } catch (IOException e) {
          // nothing to do here
        }
      }
      return null;
    }

Reply:
That would also work, do you prefer it?